src/main/java/org/apache/maven/plugins/dependency/resolvers/OldResolveDependencySourcesMojo.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugins.dependency.resolvers;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+/**
+ * Goal that resolves the project source dependencies from the repository.
+ *
+ * @author <a href="mailto:brianf@apache.org">Brian Fox</a>
+ * @since 2.0-alpha2
+ * @deprecated in favor of {@code resolve-sources} mojo
+ */
+@Mojo(
+        name = "sources",
+        defaultPhase = LifecyclePhase.GENERATE_SOURCES,
+        requiresDependencyResolution = ResolutionScope.TEST,
+        threadSafe = true)
+@Deprecated
+public class OldResolveDependencySourcesMojo extends ResolveDependencySourcesMojo {
+
+    @Override
+    protected void doExecute() throws MojoExecutionException {
+        getLog().warn(
+                        "The goal 'dependency:sources' is deprecated for removal in a future version of this plugin, please use 'dependency:resolve-sources' instead");
+        super.doExecute();
+    }
```

Comment:
We will have duble warning - one by Maven itself:

```
[WARNING]  Goal 'sources' is deprecated: in favor of resolve-sources mojo
```

and second by plugin:

```
[WARNING] The goal 'dependency:sources' is deprecated for removal in a future version of this plugin, please use 'dependency:resolve-sources' instead
```

We can extend info in docs and will be emited by Maven.

Reply:
Good catch. I will remove the log message.